spikeextractors/extractors/nwbextractors/nwbextractors.py

```diff
@@ -647,10 +647,12 @@ def add_electrical_series(recording: se.RecordingExtractor, nwbfile=None, metada
                 channel_conversion = gains * 1e-6
 
             if isinstance(recording.get_traces(), np.memmap):
+                n_bytes = np.dtype(recording.get_dtype()).itemsize
+                buffer_size = int(buffer_mb * 1e6) // (recording.get_num_channels() * n_bytes)
                 ephys_data = H5DataIO(
                     DataChunkIterator(
                         data=recording.get_traces().T,  # nwb standard is time as zero axis
-                        buffer_size=10**6
+                        buffer_size=buffer_size
```

Comment:
This ought to mimic that part of the `write_to_binary_dat_format()` function. For 500mb acting on int16, it actually solves for something about 3x larger than the 10**6, so should be a good default.

Reply:
Have you added it to the docstring?

Reply:
Nope, forgot; I'll do that now.

Reply:
OK, added to docstring; also added some minimal error checking to it as an (a) 'int' type and (b) at least 10MB. I can imagine a pathological use case where it's set to a low (but non-negative) value such that the floored division in the chunk size calculation returns zero, which would definitely be problematic.